File: rps.f90
		!This is a rock paper scissors program (fortran version)
		
		program rps
		
		implicit none
		
		character(len=10)::name
		character(len=1)::rps_choice,comp_choice1,cont
		character::allowed(6)
		character(len=1),dimension(3)::comp_choice
		real::r,a
		integer::score,played
		
		print*,"**************************************************"
		print*,"Welcome! Let's play a game of Rock Paper Scissors."
		print*,"**************************************************"
		print*," "
		print*,"First of all, what is your name?"
		
		read*,name
		
		print*,"Hi!",name
		
		score=0
		played=0
		
		30 continue
		
		print*,"Choose your weapon!"
		print*,"[R]ock, [P]aper, or [S]cissors."
		
		read*,rps_choice
		
		if(rps_choice=='r') then
			rps_choice='R'
		
		else if(rps_choice=='s') then
			rps_choice='S'
		
		else if(rps_choice=='p') then
			rps_choice='P'
		
		else
			
		100 continue
		
		end if
			
		if (any((/'R','S','P'/)==rps_choice)) then
			go to 50
			
		else
			print*,"Please key in only R for rock, P for paper,&
					&S for scissors."
			go to 30
		end if
		
		50 continue
		print*,"Your choice ",rps_choice
		comp_choice = ['R','P','S']
		
		played=played+1
		
		call random_seed()
		call random_number(r)
		
		a=r*3.0+1.0
		
		comp_choice1=comp_choice(int(a))
		
		print*,"My choice: ",comp_choice1
		
		if(rps_choice == comp_choice(int(a))) then
			
			print*,"It's a tie!"
			score=score+0
			go to 10
		
		else
			
			if(rps_choice=='R' .and. comp_choice1=='P') then
				
				print*,"Paper beats rock, I win!"
				score=score+0
				go to 10
			
			else if(rps_choice=='P' .and. comp_choice1=='S') then
				print*,"Scissors beats paper, I win!"
				score=score+0
				go to 10
				
			else if(rps_choice=='S' .and. comp_choice1=='R') then
				print*,"Rock beats scissors, I win!"
				score=score+0
				go to 10
				
			else
				print*,"Congrats! You win!"
				score=score+1
			end if

		end if
		
		10 continue
		print*,"Your current score is",score,"/",played
		print*,"Do you want to continue? [Y]es"
		
		read*,cont
		
		if(cont=='Y' .or. cont=='y') then
			go to 30
			
		else
			print*,"Congrats! You have won",score,"times out of",played,"times!"
			print*,"Thank you for playing!"
		end if
		
		end program rps

File: rps.py
#This program will allow user to play Rock, Paper, or Scissors with computer.
#We learn how to interact with the os and re modules.
#Also, we can learn how nested while loops work with proper continue or break statement.
#Overall, this is a good program for beginners.
#Credit to: https://hackr.io/blog/python-projects

import random
#import the module to interact with the operating system
import os
#import the module that will compare text strings
import re

#check the type of operating system
#if nt(windows), then 'cls' to clear screen(CMD). Else, 'clear' to clear screen.
os.system('cls' if os.name=='nt' else 'clear')

print("\n")

print("Rock, Paper, Scissors - Shoot!")

gameScore=0     #Number of wins
nTurn=0         #Number of turns

userName = input("What's your name?")

while True:
    nTurn=nTurn+1

    while True:
    #Ask user to input R, P, or S
        userChoice = input("Choose your weapon [R]ock, [P]aper, or [S]cissors: ")

    #Check if user key in any input.
        if not re.match("[SsRrPp]",userChoice):
            print("Please choose a letter: ")
            print("[R]ock, [P]aper, or [S]cissors only.")
            continue
        break
    
    #Echo the user's choice
    print(userName + " chose: " + userChoice)

    choices = ['R','P','S']

    opponentChoice = random.choice(choices)

    print("I chose: " + opponentChoice)

    if opponentChoice == str.upper(userChoice):
        gameScore=gameScore+0
        print("It's a tie!")

    elif opponentChoice=='R' and userChoice.upper()=='S':
        gameScore=gameScore+0
        print("Rock beats Scissors, I win!")

    #continue

    elif opponentChoice=='S' and userChoice.upper()=='P':
        gameScore=gameScore+0
        print("Scissors beats paper, I win!")

    #continue

    elif opponentChoice == 'P' and userChoice.upper() == 'R':
        gameScore=gameScore+0
        print("Paper beats rock, I win!")

    #continue

    else:
        gameScore=gameScore+1
        print(userName + " win!")
        
    print("Current score is " + str(gameScore) + "/" + str(nTurn))
    cont = input("Do you want to continue? Y/N  ")
    print("\n")
    
    if not(cont=='N' or cont=='n'):
    
        continue
    
    break

print("You have won "+str(gameScore)+" times!")
print("Thank you for playing!")
